### ISSUE TITLE
Broken AWS installation on "provides mysql"

### ISSUE THREAD

**Issue by shrinathaithal** [chef-cookbooks/mysql#364]
I have a cookbook which depends on this cookbook. The recent changes which brought in `provides mysql` in https://github.com/chef-cookbooks/mysql/blob/v6.0.25/libraries/provider_mysql_client.rb#L8 caused AWS opsworks deployment to break. Any known reasons? 

This was the error given by AWS Opsworks - 

```
================================================================================
Recipe Compile Error in /var/lib/aws/opsworks/cache.stage2/cookbooks/mysql/libraries/provider_mysql_client.rb
================================================================================
 
 
NoMethodError
-------------
undefined method `provides' for Chef::Provider::MysqlClient:Class
 
 
Cookbook Trace:
---------------
/var/lib/aws/opsworks/cache.stage2/cookbooks/mysql/libraries/provider_mysql_client.rb:8:in `<class:MysqlClient>'
/var/lib/aws/opsworks/cache.stage2/cookbooks/mysql/libraries/provider_mysql_client.rb:6:in `<class:Provider>'
/var/lib/aws/opsworks/cache.stage2/cookbooks/mysql/libraries/provider_mysql_client.rb:5:in `<class:Chef>'
/var/lib/aws/opsworks/cache.stage2/cookbooks/mysql/libraries/provider_mysql_client.rb:4:in `<top (required)>'
 
 
Relevant File Content:
----------------------
/var/lib/aws/opsworks/cache.stage2/cookbooks/mysql/libraries/provider_mysql_client.rb:
 
1:  require 'chef/provider/lwrp_base'
2:  require_relative 'helpers'
3:  
4:  class Chef
5:    class Provider
6:      class MysqlClient < Chef::Provider::LWRPBase
7:        include MysqlCookbook::Helpers
8>>       provides :mysql_client
9:  
10:        use_inline_resources
11:  
12:        def whyrun_supported?
13:          true
14:        end
15:  
16:        action :create do
17:          # From helpers.rb
 ```

**Comment by StarlingHealth** [comment]
I too am having this same problem.

I'm assuming it has to do with the fact that this cookbook now requires Chef 12, but AWS OpWorks only goes up to 11 for linux instances.

This is going to be a big issue for me, but I'm not sure we can ask for continues support of Chef 11?

**Comment by StarlingHealth** [comment]
If you make use of custom cookbooks, you can update your berksfile and add the following:

cookbook 'mysql', '= 6.0.24'

Which will allow you to update again, but this will lock you in to the last version of mysql.

**Comment by jorhett** [comment]
Also broken on Amazon Linux FWIW.

**Comment by shrinathaithal** [comment]
Sadly, this cookbook is not directly referenced by me, it is in one of the dependencies of another third-party cookbook which `depends` on this. Any ideas how to stick all of these dependencies to 6.0.24? 
I tried mentioning this cookbook with version in Berkshelf file before/after adding 3rd-party cookbook, but it still downloaded latest one :(

**Comment by jorhett** [comment]
Why does reverting to 6.0.24 fix the problem then?

**Comment by StarlingHealth** [comment]
In my case it works because I have a direct dependency and am calling down just that one version. In shrinathaithal case they have a dependency that is specifying the latest as being required, and it wont run. Even if you spec the lower version, the dependency will pull down the latest.

**Comment by someara** [comment]
6.0.25 reverted the changes in 6.0.24.
This means you still need to pin Chef belong 12.4.0.

Working on 6.0.26 tonight =)

-s

**Comment by someara** [comment]
This is fixed as of v6.0.28.
Also, 12.4.1 should have fixed the problem with 6.0.24 as well.

-s
